### DISCUSSION TITLE
Handle Light and Dark themes

### DISCUSSION THREAD

**Discussion by 0xfalafel** [ratatui/ratatui#1401] [Discussion]
So, I have created an application, and during the development I have __focused__ mainly on __dark themes__.
It is usable with a light theme, but arguably worst.

![hextazy_black_and_white2](https://github.com/user-attachments/assets/4546f469-021b-47c4-b14a-1d584c3a5cb7)

I understand that part of the solution is picking colors that work with any terminal.

But I was wondering if it is possible to:
*  __detect__ if the __theme is light or dark__.
*  __get__ the __default colors__ for the borders, background, and characters.

For example, here, I think it would be more readable if the *character* selected by the cursor was of the color of the background.
I.e., __`7`__ fg color would be __white__ (default color of the background) on the __light terminal__.

I am also curious to have your opinion. Like if this is the wrong strategy for these issues.

**Comment by kdheepak** [comment]
Good question! (hextazy looks great btw)

In `crates-tui`, we made a color scheme file that users could read in. And they were based on base16 values - https://github.com/ratatui/crates-tui/blob/main/.config/dracula.yaml 

You can see roughly what colors neovim and different editors use for the different UI elements - https://github.com/ratatui/crates-tui/blob/2bab7c1b3ead79b00bfbe1444f4571e08b38f251/src/config.rs#L19-L85

I made a rust crate to pull this `Base16` struct out of `crates-tui` - https://github.com/kdheepak/ratatui-base16

**Comment by joshka** [comment]
> **detect** if the **theme is light or dark**.

If you're after detecting whether the terminal is in dark or light mode, look at:
- https://crates.io/crates/dark-light
- https://crates.io/crates/termbg

**Comment by orhun** [comment]
You can also take a look at my implementation for [`binsider`](https://binsider.dev/): https://github.com/orhun/binsider/pull/23 (selects the accent color based on the terminal background)

Some info about how `termbg` works: https://github.com/dalance/termbg/pull/24#issuecomment-2354445187

**Reply by orhun** [reply]
Also, here is another example about adding theme options to your application: https://github.com/orhun/flawz/commit/4f669746146c216becb3f2ac2c1d1c474ce20511
